sign/ed25519/ed25519.go

```diff
@@ -1,16 +1,26 @@
+// Package ed25519 implements Ed25519 signature scheme as described in RFC-8032.
+//
+// References:
+//  - RFC8032 https://rfc-editor.org/rfc/rfc8032.txt
+//  - Ed25519 https://ed25519.cr.yp.to/
+//  - High-speed high-security signatures. https://doi.org/10.1007/s13389-012-0027-1
 package ed25519
 
 import (
 	"bytes"
 	"crypto"
-	"crypto/rand"
+	cryptoRand "crypto/rand"
 	"crypto/sha512"
 	"errors"
 	"io"
 )
 
-// Size is the length in bytes of Ed25519 keys.
-const Size = 32
+const (
+	// Size is the length in bytes of Ed25519 keys.
+	Size = 32
+	// SignatureSize is the length in bytes of signatures.
+	SignatureSize = 2 * Size
+)
 
 // PublicKey represents a public key of Ed25519.
 type PublicKey []byte
```

Comment:
Why not `*[Size]byte`?

Reply:
Same as in my comment above. I agree that a constant-sized array is better. Also, how does a user of CIRCl interoperates between ed25519 and ed448 signatures?